ci-operator/step-registry/rosa/nodes/wait-ready/rosa-nodes-wait-ready-commands.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
+
+ENABLE_AUTOSCALING=${ENABLE_AUTOSCALING:-false}
+CLUSTER_ID=$(cat "${SHARED_DIR}/cluster-id")
+echo "CLUSTER_ID is $CLUSTER_ID"
+
+if [[ "$ENABLE_AUTOSCALING" == "true" ]]; then
+    desired_node_count=$(rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r '.nodes.autoscale_compute.min_replicas')
+else
+    desired_node_count=$(rosa describe cluster -c "$CLUSTER_ID"  -o json  |jq -r '.nodes.compute')
+fi
+echo "Desired node count: $desired_node_count"
+
+# wait for all nodes to reach Ready=true to ensure that all machines and nodes came up, before we run
+# any e2e tests that might require specific workload capacity.
+echo "$(date) - waiting for worker nodes to be ready..."
+ret=0
+oc wait nodes --all --for=condition=Ready=true --timeout=10m || ret=$?
+if [[ "$ret" == 0 ]]; then
+    node_count="$(oc get nodes --no-headers -l node-role.kubernetes.io/worker,node-role.kubernetes.io/SchedulingDisabled!=,node-role.kubernetes.io/infra!=,node-role.kubernetes.io/workload!= | wc -l)"
+    echo "Count of nodes in Ready state: $node_count"
+    if [[ "$ENABLE_AUTOSCALING" == "false" ]]; then
+        # Check worker node count matches requested REPLICAS
+        if [[ "$node_count" == "$desired_node_count" ]]; then
+            echo "$(date): All $node_count worker nodes are ready and match desired $desired_node_count replica count"
+        fi
+    else
+        # Check and confirm node count is no less than MIN_REPLICAS.
+        if (( "$node_count" >= "$desired_node_count" )); then
+            echo "$(date): All $node_count worker nodes are ready and match desired $desired_node_count min_replicas node count"
+        fi
```

Comment:
We do not need to separate it by  `ENABLE_AUTOSCALING`. `"$node_count" >= "$desired_node_count"` is enough for all of the conditions.